### ISSUE TITLE
startChat not respecting name and email on Android

### ISSUE THREAD

**Issue by diogoviannaaraujo** [taskrabbit/react-native-zendesk-chat#65]
Hi, we have a app that is being built for iOS and Android, on iOS I've been able to set user name and email so I can identify him on Zendesk chat dashboard, but on android the user is always handled as a Visitor.

Been calling startChat like:
```
        onPress: () =>
          ZendeskChat.startChat({
            department: 'App',
            localizedDismissButtonTitle: 'Voltar',
            name: 'User Name',
            email: 'user@email.com',
          })
```

The department is correctly selected, but on Android the user is always a visitor.

**Comment by fbartho** [comment]
It looks like you have the right code @diogoviannaaraujo -- I don't know what's going on?

**Comment by fbartho** [comment]
If the builder properly has your data in it, then I'd check with Zendesk Support to see what the problem might be.

**Comment by diogoviannaaraujo** [comment]
Ill be looking on that tomorrow, i had once or twice seen on iOS the bot responding with a visitor name on greeting and then changing to correct name, so maybe its a zendesk glitch.

**Comment by jakmak24** [comment]
Looks like a bug in Android SDK, preform is overriding visitorInfo settings https://develop.zendesk.com/hc/en-us/community/posts/360044257813/comments/360011630833

**Comment by fbartho** [comment]
@jakmak24 that sounds like exactly the root cause.

@diogoviannaaraujo -- do you still need this ticket, since the issue is a difference in behavior in Zendesk's libraries?

**Comment by diogoviannaaraujo** [comment]
Hey, no need to. Will try to get around it then.

**Comment by fbartho** [comment]
@diogoviannaaraujo this was I had put in our app:

```js
			preChatFormOptions: {
				name: !name ? "required" : "optional", // Notice this line -- you could use "none" instead!
				email: "optional",
				phone: "optional",
				department: "required",
			},
```
Additionally, you can disable the preChatForm in `behaviorFlags` if you detect you don't need the Form.

Please let me know if you need further help!

**Comment by firdaussoberi** [comment]
I know this is closed, but I'm experiencing same issue with v0.4.0 of the library. RN0.60.4.
Setting like this, first the preform bot shows up and ask name, email and phone. Only then in dashboard the info shows up.
How do I preset the details from login info in my app? I did like this but even explicitly presetting won't make it work...
```javascript
       onPress: () =>
          ZendeskChat.startChat({
            department: 'appname',
            localizedDismissButtonTitle: 'Dismiss',
            name: 'user',
            email: 'user@email.com',
            phone: '12345',
          });
```

**Comment by firdaussoberi** [comment]
Using 0.4.1-beta.4 fixed this. This will be merged into v0.4.1 soon right? Thanks for the great libary!

**Comment by fbartho** [comment]
Indeed, the the beta has passed QA at a couple companies, and we'll probably ship it in production in our next release.

I don't know when I'll get a chance to do the full official npm release without the beta tag though!
